dive_core/command_hierarchy.cpp

```diff
@@ -2089,32 +2093,157 @@ void CommandHierarchyCreator::AppendPacketFieldNodes(const IMemoryManager &mem_m
     }
 
     // If there are missing packet fields, then output the raw DWORDS directly
-    if (end_dword < type7_header.count)
+    // Some packets, such as CP_LOAD_STATE6_* handle this explicitly elsewhere
+    if (append_extra_dwords)
     {
-        for (size_t i = end_dword + 1; i <= type7_header.count; i++)
+        if (end_dword < dword_count)
         {
-            uint32_t dword_value = 0;
-            uint64_t dword_va_addr = va_addr + i * sizeof(uint32_t);
-            bool     ret = mem_manager.CopyMemory(&dword_value,
-                                              submit_index,
-                                              dword_va_addr,
-                                              sizeof(uint32_t));
-            DIVE_VERIFY(ret);  // This should never fail!
-
-            std::ostringstream field_string_stream;
-            field_string_stream << "(DWORD " << i << "): 0x" << std::hex << dword_value;
+            for (size_t i = end_dword + 1; i <= dword_count; i++)
+            {
+                uint32_t dword_value = 0;
+                uint64_t dword_va_addr = va_addr + i * sizeof(uint32_t);
+                bool     ret = mem_manager.CopyMemory(&dword_value,
+                                                  submit_index,
+                                                  dword_va_addr,
+                                                  sizeof(uint32_t));
+                DIVE_VERIFY(ret);  // This should never fail!
+
+                std::ostringstream field_string_stream;
+                field_string_stream << prefix << "(DWORD " << i << "): 0x" << std::hex
+                                    << dword_value;
+
+                CommandHierarchy::AuxInfo aux_info = CommandHierarchy::AuxInfo::RegFieldNode(false);
+                uint64_t                  field_node_index = AddNode(NodeType::kFieldNode,
+                                                    field_string_stream.str(),
+                                                    aux_info);
+
+                // Add it as child to packet_node
+                AddChild(CommandHierarchy::kEngineTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kSubmitTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kAllEventTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kRgpTopology, packet_node_index, field_node_index);
```

Comment:
not specific to your change, but i think we should remove those unused topology asap. i guess `kRgpTopology` is not used for sure, how about `kEngineTopology` and `kSubmitTopology`?

Reply:
kEngineTopology is used if we have different queues. It's a queue-centric view. But yes, at the very very minimum, we shouldn't call it an "engine" the way AMD driver does. But I'm leaning towards removing it altogether.

I'm definitely using submit topology. When you select the "Submit" mode, that's what you're seeing.

Reply:
what is the difference between `kSubmitTopology` and `kAllEventTopology`?

Reply:
Submit topology lets you see a list of IBs. All-Event lets you see a list of all the GPU events (draws, dmas, etc).

Reply:
ah, i see, thanks!